Public/Src/Engine/Scheduler/Tracing/Log.cs

```diff
@@ -2596,6 +2596,38 @@ internal void FailedToHashInputFile(LoggingContext loggingContext, string pipDes
             string pipDescription,
             string pipValueId);
 
+        [GeneratedEvent(
+            (int)EventId.ScheduleFailAddPipDueToInvalidPreserveOutputWhitelist,
+            EventGenerators = EventGenerators.LocalOnly,
+            EventLevel = Level.Error,
+            Keywords = (int)(Events.Keywords.UserMessage | Events.Keywords.UserError),
+            EventTask = (int)Events.Tasks.Scheduler,
+            Message = "The pip '{pipDescription}' could not be added because one of PreserveOutputWhitelist is neither static file output nor directory output).")]

```

Comment:
> could not be added because one of PreserveOutputWhitelist is neither static file output nor directory output [](start = 49, length = 109)

nit - suggest rewording to somethign like:
could not be added because it specified a PreserveOutputWhitelist which included {path} which was not defined as one of the pip's file or directory outputs.